kubernetes-client/src/main/java/io/micronaut/kubernetes/client/DiscoveryCache.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client;
+
+import io.kubernetes.client.Discovery;
+import io.kubernetes.client.apimachinery.GroupVersionKind;
+import io.kubernetes.client.common.KubernetesObject;
+import io.kubernetes.client.openapi.ApiException;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.context.annotation.Value;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Bean that provides caching over the {@link Discovery} results.
+ *
+ * @author Pavol Gressa
+ * @since 3.1
+ */
+@Requires(beans = Discovery.class)
+@Singleton
+public class DiscoveryCache {
+    private static final Logger LOG = LoggerFactory.getLogger(DiscoveryCache.class);
+
+    private static final ModelMapper MODEL_MAPPER = new ModelMapper();
+
+    private final Discovery discovery;
+    private final Duration refreshInterval;
+
+    private Set<Discovery.APIResource> lastAPIDiscovery = new HashSet<>();
+    private volatile long nextDiscoveryRefreshTimeMillis = 0;
+
+    public DiscoveryCache(Discovery discovery,
+                          @Value("${kubernetes.client.api-discovery.cache.refresh-interval:30}") long discoveryRefreshInterval) {
```

Comment:
Again can this be supplied by configuration properties?

Reply:
ah, why not, but there will be just one property